src/main/java/com/github/cameltooling/lsp/internal/parser/ParserXMLFileHelper.java

```diff
@@ -42,13 +42,15 @@
 	private static final String ATTRIBUTE_ROUTE = "route";
 	private static final String NAMESPACEURI_CAMEL_BLUEPRINT = "http://camel.apache.org/schema/blueprint";
 	private static final String NAMESPACEURI_CAMEL_SPRING = "http://camel.apache.org/schema/spring";
-
+	private static final String URI_PARAM = "uri=\"";
+	
 	public String getCamelComponentUri(String line, int characterPosition) {
-		int uriAttribute = line.indexOf("uri=\"");
+		int uriAttribute = line.indexOf(URI_PARAM);
 		if(uriAttribute != -1) {
-			int nextQuote = line.indexOf('\"', uriAttribute + 5);
-			if (isBetween(characterPosition, uriAttribute + 5, nextQuote)) {
-				return line.substring(uriAttribute + 5, nextQuote);
+			int firstQuote = line.indexOf('\"', uriAttribute);
```

Comment:
this is dangerous, if you have another attribute on the same line, for instance the id before, it won't work. you need to keep the search on uri=\"

if it has not break it worth adding a test for this case (or at least to raise an issue to add it)

Reply:
I don't think you are right. I am starting to search in the line at the position of the URI attribute which will most likely be followed by ="value".

Reply:
my bad, you're right